main.cpp

```diff
@@ -122,15 +122,48 @@ void mettreAJourContact(std::vector<Contact>& contacts) {
 
 void supprimerContact(std::vector<Contact>& contacts) {
     std::string numero;
-    std::cout << "Entrez le num�ro de telephone du contact a supprimer : ";
+    std::cout << "Entrez le num�ro de t�l�phone du contact � supprimer : ";
     std::getline(std::cin, numero);
 
     for (auto it = contacts.begin(); it != contacts.end(); ++it) {
         if (it->numero_telephone == numero) {
             contacts.erase(it);
```

Comment:
La suppression des éléments dans un vecteur en utilisant un for et erase peut être simplifiée avec std::remove_if pour rendre l'opération plus claire et efficace.